include/blas_meta.h

```diff
@@ -222,6 +222,18 @@ struct is_complex_std
 
 #endif
 
+class unsupported_exception : public std::runtime_error {
+ public:
+  unsupported_exception(const std::string &operator_name)
+      : std::runtime_error(operator_name), _msg(operator_name) {
+    _msg += " operator currently not supported on Arc or GPU Max GPUs";
+  };
+  const char *what() const throw() { return _msg.c_str(); }
```

Comment:
Should this be marked as virtual?

Reply:
I don't think we should/will ever derive from this exception class, so if it is `virtual` or not doesn't matter.

Reply:
I think it should be indicated that its virtual in anyway using `override` - [C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-override). You'll also get warnings in oneMKL without this - see my previous [PR](https://github.com/oneapi-src/oneMKL/pull/454).

Reply:
Thank you for the reference links, fixed in [75960e5](https://github.com/codeplaysoftware/portBLAS/pull/524/commits/75960e5bd3b523c9850d21feb017f6b3523b5595)